org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/ast/AbstractVariableDeclaration.java

```diff
@@ -144,4 +145,15 @@ public void setDepth(int depth) {
 	public void setFieldIndex(int depth) {
 		// do nothing by default
 	}
+
+	/**
+	 * Returns true if this variable is an unnamed variable (_) and false otherwise.
+	 *
+	 * @param scope used to determine source level
+	 */
+	public boolean isUnnamed(BlockScope scope) {
+		return ((this.name.length == 1 && this.name[0] == '_') || ("\\u005F".equals(this.name.toString()))) //$NON-NLS-1$
+				&& scope.compilerOptions().sourceLevel >= ClassFileConstants.JDK21
+				&& scope.compilerOptions().enablePreviewFeatures;
```

Comment:
Curious - is the ` ("\\u005F".equals(this.name.toString())))` really required ? Won't the scanner internalize the unicode and return an '_' ??